### DISCUSSION TITLE
How can I avoid duplicate form elements using MultiSelect?

### DISCUSSION THREAD

**Discussion by NicolasSacC** [filamentphp/filament#4145] [Help] [answered]
Hi!

Don't know if it's a feature or a bug:

I create a MultiSelect using a Closure. I get only the models that make sense for my request:
```
                        Forms\Components\MultiSelect::make('edit_requests')
                            ->label('Demandes de ramassage disponibles')
                            ->options(
                                fn (?Model $record) => PickupRequest::active()->where(function ($query) use ($record) {
                                    $query->whereDoesntHave('route')->orWhere('route_id', $record->id);
                                })->get()->pluck('uniqid', 'id')->unique()
                            )
                            ->visibleOn('edit'),

```

Before filling the form, I edit the data to include the already selected fields:
```
    protected function mutateFormDataBeforeFill(array $data): array
    {
        $data['edit_requests'] = $this->record->requests()->pluck('id')->toArray();

        return $data;
    }
```

However, I get duplicates if it is already selected:
![image](https://user-images.githubusercontent.com/16766008/191792054-5f313c56-a6b4-417c-b657-ab5b264cdb02.png)

Any idea what might be the issue here? I feel like the data I pass is like creating new fields instead of selecting the ones that are already there... Maybe because of the Closure?

**Comment by danharrin** [comment, marked as answer]
It's a bug - #2363
